File: AFMobile/AFAndroid/app/src/main/java/cz/cvut/fel/matyapav/afandroid/components/parts/validators/MaxValueValidator.java
package cz.cvut.fel.matyapav.afandroid.components.parts.validators;

import android.content.Context;
import android.widget.EditText;

import cz.cvut.fel.matyapav.afandroid.components.parts.AFField;
import cz.cvut.fel.matyapav.afandroid.components.parts.ValidationRule;
import cz.cvut.fel.matyapav.afandroid.utils.Localization;
import cz.cvut.fel.matyapav.afandroid.utils.Utils;

/**
 * Validator which checks if input value is smaller than maximum value
 *
 * @author Pavel Matyáš (matyapav@fel.cvut.cz).
 * @since 1.0.0..
 */
public class MaxValueValidator implements AFValidator {

    @Override
    public boolean validate(Context context, AFField field, StringBuilder errorMsgs, ValidationRule rule) {
        boolean validationIsFine = true;
        if(Utils.isFieldNumberField(field)){
            EditText numberField = (EditText) field.getFieldView();
            if(numberField.getText() != null && !numberField.getText().toString().isEmpty() &&
                    Double.valueOf(numberField.getText().toString()) > Double.valueOf(rule.getValue())){
                validationIsFine = false;
            }
        }
        if(!validationIsFine){
            errorMsgs.append(Localization.translate(context, "validation.maxval")).append(" ").append(rule.getValue());
        }
        return validationIsFine;
    }
}

File: AFMobile/AFWindowsPhone/AFWindowsPhone/components/parts/validators/MaxValueValidator.cs
using AFWindowsPhone.utils;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Windows.UI.Xaml.Controls;
using AFWindowsPhone.components.parts.validators;

namespace AFWindowsPhone.builders.components.parts.validators
{
    class MaxValueValidator : AFValidator
    {
        public bool validate(AFField field, StringBuilder errorMsgs, ValidationRule rule)
        {
            bool validationIsFine = true;
            if (Utils.IsFieldNumberField(field))
            {
                TextBox numberField = (TextBox)field.getFieldView();
                try
                {
                    if (!String.IsNullOrEmpty(numberField.Text) &&
                        Convert.ToDouble(numberField.Text) > Convert.ToDouble(rule.getValue()))
                    {
                        validationIsFine = false;
                    }
                }
                catch (Exception ex) //catch convert exception
                {
                    validationIsFine = true; //if there is no number in field we cant validate its value
                }
            }
            if (!validationIsFine)
            {
                errorMsgs.Append(Localization.translate("validation.maxval") + " " + rule.getValue());
            }
            return validationIsFine;

        }
    }
}
